x-pack/plugins/apm/server/routes/metrics/by_agent/serverless/active_instances.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { termQuery } from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { euiLightVars as theme } from '@kbn/ui-theme';
+import {
+  FAAS_ID,
+  METRICSET_NAME,
+  SERVICE_NAME,
+  SERVICE_NODE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../../../common/utils/environment_query';
+import { getVizColorForIndex } from '../../../../../common/viz_colors';
+import { getMetricsDateHistogramParams } from '../../../../lib/helpers/metrics';
+import { Setup } from '../../../../lib/helpers/setup_request';
+import { GenericMetricsChart } from '../../fetch_and_transform_metrics';
+
+export async function getActiveInstances({
+  environment,
+  kuery,
+  setup,
+  serviceName,
+  start,
+  end,
+  faasId,
+}: {
+  environment: string;
+  kuery: string;
+  setup: Setup;
+  serviceName: string;
+  start: number;
+  end: number;
+  faasId?: string;
+}): Promise<GenericMetricsChart> {
+  const { apmEventClient, config } = setup;
+
+  const aggs = {
+    activeInstances: {
+      cardinality: {
+        field: SERVICE_NODE_NAME,
+      },
+    },
+  };
+
+  const params = {
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
+            ...rangeQuery(start, end),
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            { term: { [METRICSET_NAME]: 'transaction' } },
```

Comment:
Why don't we have the switch here to toggle between transaction metrics and events as we commonly do for other transaction-based data?